pkg/controllers/generic/export_syncer.go

```diff
@@ -243,6 +249,9 @@ func (f *exporter) Sync(ctx *synccontext.SyncContext, pObj client.Object, vObj c
 
 			return ctrl.Result{}, nil
 		}
+		if kerrors.IsConflict(err) {
+			return ctrl.Result{Requeue: true}, nil
+		}
```

Comment:
@FabianKramm please double check this one. Above you have `ReplaceOnConflict`? Is that intentional and correct?

Reply:
@sttts thanks for the pointer, the `IsInvalid` is definitely required for resources that have immutable fields and need to be replaced if updated on one side as otherwise this will get stuck in an error retry loop.

You probably know this better than me, but are there any other conflict errors in Kubernetes except the `the object has been modified; please apply your changes to the latest version and try again`? If yes, I believe we should actually check its this exact error instead of just checking for if its a conflict error

Reply:
A patch or update will fail if some precondition (resource version, uid) does not hold. For a patch in controller-runtime it's usually `client.MergeFromWithOptimisticLock{}` as patch parameter. Both cases should be fixed through a requeue. `IsInvalid` is different maybe because multiple actors could get stuck when applying changes from both sides. With replace you can then force your view of the world. I guess that's the purpose of the code here.

Reply:
@sttts that makes sense, thanks for the explanation! I think we should remove the `kerrors.IsConflict` then from the condition above and just leave `kerrors.IsInvalid`

Reply:
done